File: CSharp/cs_EasyMSX-master/EasyMSX/Teams.cs

using System;
using System.Collections;
using System.Collections.Generic;

using LogLevels = com.bloomberg.emsx.samples.Log.LogLevels;
using Element =  Bloomberglp.Blpapi.Element;
using Message = Bloomberglp.Blpapi.Message;
using Name = Bloomberglp.Blpapi.Name;
using Request = Bloomberglp.Blpapi.Request;

namespace com.bloomberg.emsx.samples {

    public class Teams : IEnumerable<Team> {

        private static readonly Name    GET_TEAMS = new Name("GetTeams");
	    private static readonly Name 	ERROR_INFO = new Name("ErrorInfo");

	    private List<Team> teams = new List<Team>(); 
        internal EasyMSX emsxapi;
    
	    internal Teams(EasyMSX emsxapi) {
		    this.emsxapi = emsxapi;
		    loadTeams();
	    }
	
	    private void loadTeams() {
		    Log.LogMessage(LogLevels.BASIC, "Teams: Loading");
		    Request request = emsxapi.emsxService.CreateRequest(GET_TEAMS.ToString());
            emsxapi.submitRequest(request,new TeamHandler(this));
	    }

	    class TeamHandler : MessageHandler {
		
		    Teams teams;
		
		    internal TeamHandler(Teams teams) {
			    this.teams = teams;
		    }
		
		    public void processMessage(Message message) {
			
			    Log.LogMessage(LogLevels.BASIC, "Teams: processing message");
			
	    	    if(message.MessageType.Equals(ERROR_INFO)) {
	        	    Log.LogMessage(LogLevels.BASIC, "Teams: processing RESPONSE error");
	    		    int errorCode = message.GetElementAsInt32("ERROR_CODE");
	    		    String errorMessage = message.GetElementAsString("ERROR_MESSAGE");
	    		    Log.LogMessage(LogLevels.BASIC, "Error getting teams: [" + errorCode + "] " + errorMessage);
	    	    } else if(message.MessageType.Equals(GET_TEAMS)) {
	        	    Log.LogMessage(LogLevels.BASIC, "Teams: processing successful RESPONSE");
	
	        	    Element teamList = message.GetElement("TEAMS");
	    		
				    int numValues = teamList.NumValues;
	    		
	    		    for(int i = 0; i < numValues; i++) {
	    			    String teamName = teamList.GetValueAsString(i);
	    			    Team newTeam = new Team(teams,teamName);
	    			    teams.add(newTeam);
	            	    Log.LogMessage(LogLevels.DETAILED, "Teams: Added new team " + newTeam.name);
	    		    }
	    	    }
		    }
	    }

        public IEnumerator<Team> GetEnumerator()
        {
            return teams.GetEnumerator();
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return this.GetEnumerator();
        }
	
	    public Team get(int index) {
		    return teams[index];
	    }
	
	    public Team get(String name) {
		    foreach(Team t in teams) {
			    if(t.name.Equals(name)) return t;
		    }
		    return null;
	    }

	    private void add(Team newTeam) {
		    teams.Add(newTeam);
	    }
    }
}

File: Python/easymsx-1.0.1/easymsx/teams.py
# teams.py

import blpapi
from .team import Team

GET_TEAMS = blpapi.Name("GetTeams")
ERROR_INFO = blpapi.Name("ErrorInfo")

class Teams:
    
    def __init__(self,easymsx):
        self.easymsx = easymsx
        self.teams=[]
        self.load_teams()

    def __iter__(self):
        return self.teams.__iter__()

    def load_teams(self):
        
        request = self.easymsx.emsx_service.createRequest(str(GET_TEAMS))
        self.easymsx.submit_request(request, self.process_message)
        
    def process_message(self, msg):
        
        if msg.messageType() == ERROR_INFO:
            error_code = msg.getElementAsInteger("ERROR_CODE")
            error_message = msg.getElementAsString("ERROR_MESSAGE")
            print("GetTeams >> ERROR CODE: %d\tERROR MESSAGE: %s" % (error_code,error_message))

        elif msg.messageType() == GET_TEAMS:
            
            tms = msg.getElement("TEAMS")
            
            for t in tms.values():
                self.teams.append(Team(self,t))
                
    def get(self,team_name):
        for t in self.teams:
            if t.name == team_name:
                return t
        
        return None
                                  
__copyright__ = """
Copyright 2017. Bloomberg Finance L.P.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to
deal in the Software without restriction, including without limitation the
rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
sell copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:  The above
copyright notice and this permission notice shall be included in all copies
or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
IN THE SOFTWARE.
"""
